doc/transceiver-monitor-hld.md

```diff
@@ -0,0 +1,152 @@
+The requirement is described in [Sensor and Transceiver Info Monitoring Requirement.](https://github.com/Azure/SONiC/blob/gh-pages/doc/OIDsforSensorandTransciver.MD)
+
+
+## 1. Xcvrd design ##
+
+New Xcvrd in platform monitor container is designed to fetch the transceiver and DOM sensor information from the eeprom and then update the state db with these info. 
+
+For the transceiver it's self, the type, serial number, hardware version, etc. will not change after plug in. The suitable way for transceiver information update can be triggered by transceiver plug in/out event.
+
+The transceiver dom sensor information(temperature, power,voltage, etc.) can change frequently, these information need to be updated periodically, for now the time period temporarily set to 60s(see open question 1), this time period need to be adjusted according the later on test on all vendors platform.
+
+If there is transceiver and sensor status change, Xcvrd will write the new status to state DB, to store these information some new tables will be added to STATE_DB.
+ 
+### 1.1 State DB Schema ###
+
+New Transceiver info table and transceiver DOM sensor table will be added to state DB to store the transceiver and DOM sensor information. 
+
+#### 1.1.1 Transceiver info Table ####
+
+	; Defines Transceiver information for a port
+	key                     = TRANSCEIVER_INFO|ifname         ; configuration for watchdog on port
+	; field                 = value
+	type                    = 1*255VCHAR                       ; type of sfp
+	hardwarerev             = 1*255VCHAR                       ; hardware version of sfp
+	serialnum               = 1*255VCHAR                       ; serial number of the sfp
+	manufacturename         = 1*255VCHAR                       ; sfp venndor name
+	modelname               = 1*255VCHAR                       ; sfp model name
+
+#### 1.1.2 Transceiver DOM sensor Table ####
+
+	; Defines Transceiver DOM sensor information for a port
+	key                     = TRANSCEIVER_DOM_SENSOR|ifname        ; configuration for watchdog on port
+	temperature             = FLOAT                                      ; temperature value in Celsius
+	voltage                 = FLOAT                                      ; voltage value
+	rx1power                = FLOAT                                      ; rx1 power in dbm
+	rx2power                = FLOAT                                      ; rx2 power in dbm
+	rx3power                = FLOAT                                      ; rx3 power in dbm
+	rx4power                = FLOAT                                      ; rx4 power in dbm
+	tx1bias                 = FLOAT                                      ; tx1 bias in mA
+	tx2bias                 = FLOAT                                      ; tx2 bias in mA
+	tx3bias                 = FLOAT                                      ; tx3 bias in mA
+	tx4bias                 = FLOAT                                      ; tx4 bias in mA
+

```

Comment:
Is it possible to add TX power? Or TX bias is enough?

Reply:
@yongcanwang00 this is defined in the requirement doc, I would like to ask the requirement author @hui-ma to comment on this.

Reply:
TX bias is from the output of "show interfaces transceiver eeprom" command, It is also the output of some other vendors' snmp output. I think they can be translated into TX power. I need to look for its equation. Do you prefer TX power in the output directly? 

    ChannelMonitorValues:
              RX1Power: -1.1936dBm
              RX2Power: -1.1793dBm
              RX3Power: -0.9388dBm
              RX4Power: -1.0729dBm
              TX1Bias: 4.0140mA
              TX2Bias: 4.0140mA
              TX3Bias: 4.0140mA
              TX4Bias: 4.0140mA
        ModuleMonitorValues :
                Temperature : 1.1111C
                Vcc : 0.0000Volts

Reply:
Yes, TX power is useful.  It's helpful to have them both.